zerver/lib/actions.py

```diff
@@ -3465,6 +3465,28 @@ def send_peer_subscriber_events(
         altered_user_ids = altered_user_dict[stream_id]
         peer_user_ids = private_peer_dict[stream_id] - altered_user_ids
 
+        # For sending user left stream messages
+        if op == "peer_remove":
+            notifications = []
+            for user_id in altered_user_ids:
+                try:
+                    user_profile = get_user_by_id_in_realm_including_cross_realm(user_id, realm)
```

Comment:
This get function makes db query and we might not want to do it within a loop (as round-trips to the database slow everything down). Instead, please consider fetching all the user profiles outside the loop.